src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -121,8 +138,50 @@ public void deleteRecording(ConnectionDescriptor connectionDescriptor, String re
                 });
     }
 
+    public List<Map<String, String>> getRecordings() throws Exception {
+        if (!fs.exists(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "does not exist");
+        }
+        if (!fs.isReadable(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "is not readable");
+        }
+        if (!fs.isDirectory(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "is not a directory");
+        }
+        WebServer webServer = webServerProvider.get();
+        List<String> files = this.fs.listDirectoryChildren(recordingsPath);
+        return files.stream()
+                .map(
+                        file -> {
+                            String serviceUriHash = file.split("/")[0];
+                            String recordingName = file.split("/")[1];
```

Comment:
Can this be done using the `Path` class rather than `split("/")`? Something like `serviceUriHash = Path.of(file).getParent()`, `recordingName = Path.of(file).getFileName()`.